src/App.js

```diff
@@ -353,9 +364,26 @@ function AppHeaderUser({
 }
 
 function FullApp() {
+  const [loggedInTracked, setLoggedInTracked] = useState(false);
+  const { trackLogin } = useContext(AnalyticsContext);
+
   const exchangeRef = useRef();
   const { connector, library, deactivate, activate, active, account } = useWeb3React();
   const { chainId } = useChainId();
+  const readerAddress = getContract(chainId, "Reader");
+  const tokens = getTokens(chainId);
+  const tokenAddresses = tokens.map((token) => token.address);
+
+  const { data: tokenBalances } = useSWR(
+    [`GlpSwap:getTokenBalances:${active}`, chainId, readerAddress, "getTokenBalances", account || PLACEHOLDER_ACCOUNT],
+    {
+      fetcher: fetcher(library, ReaderV2, [tokenAddresses]),
+    }
+  );
+  const { infoTokens } = useInfoTokens(library, chainId, active, tokenBalances, undefined);
```

Comment:
This is a round about way to get the eth balance. You can use `library.getBalance(account)` from web3React